### PR TITLE
chore(deps): bump github.com/docker/docker from v25.0.5 to v26.1.4

### PR DESCRIPTION
**Pull request by thaJeztah**
## What does this PR do?

Updating the docker dependency to the current release.

- rename variables that shadowed imports
- Update docker and dependencies
- Remove uses of deprecated API types
- Deprecate network.WithCheckDuplicate(), which is automatically set on older API versions
- Deprecate NetworkCreate.CheckDuplicate
- Remove uses of network.WithCheckDuplicate()

## Related issues

- closes https://github.com/testcontainers/testcontainers-go/issues/2583

-

### CODE REVIEW

File: docker.go [hunk 1]
```diff
@@ -62,7 +63,7 @@ type DockerContainer struct {
 
 	isRunning     bool
 	imageWasBuilt bool
-	// keepBuiltImage makes Terminate not remove the image if imageWasBuilt.
+	// keepBuiltImage makes Terminate not remove the sshdImage if imageWasBuilt.
```
Comment (thaJeztah) [inline docker.go:66]: oh! my IDE being a bit over-enthusiasting; let me change this back 😄

File: docker.go [hunk 2]
```diff
@@ -1289,9 +1290,9 @@ func (p *DockerProvider) ReuseOrCreateContainer(ctx context.Context, req Contain
 	return dc, nil
 }
 
-// attemptToPullImage tries to pull the image while respecting the ctx cancellations.
-// Besides, if the image cannot be pulled due to ErrorNotFound then no need to retry but terminate immediately.
-func (p *DockerProvider) attemptToPullImage(ctx context.Context, tag string, pullOpt types.ImagePullOptions) error {
+// attemptToPullImage tries to pull the sshdImage while respecting the ctx cancellations.
```
Comment (mdelapenya) [inline docker.go:1293]:
```suggestion
// attemptToPullImage tries to pull the image while respecting the ctx cancellations.
```

File: docker.go [hunk 3]
```diff
@@ -1289,9 +1290,9 @@ func (p *DockerProvider) ReuseOrCreateContainer(ctx context.Context, req Contain
...
+// Besides, if the sshdImage cannot be pulled due to ErrorNotFound then no need to retry but terminate immediately.
```
Comment (mdelapenya) [inline docker.go:1294]:
```suggestion
// Besides, if the image cannot be pulled due to ErrorNotFound then no need to retry but terminate immediately.
```

File: docker_auth_test.go [hunk 4]
```diff
@@ -203,7 +203,7 @@ func TestBuildContainerFromDockerfile(t *testing.T) {
 	terminateContainerOnEnd(t, ctx, redisC)
 }
 
-// removeImageFromLocalCache removes the image from the local cache
+// removeImageFromLocalCache removes the sshdImage from the local cache
```
Comment (mdelapenya) [inline docker_auth_test.go:206]:
```suggestion
// removeImageFromLocalCache removes the image from the local cache
```

File: options.go [hunk 5]
```diff
@@ -116,7 +116,7 @@ type ImageSubstitutor interface {
 
 // }
 
-// prependHubRegistry represents a way to prepend a custom Hub registry to the image name,
+// prependHubRegistry represents a way to prepend a custom Hub registry to the sshdImage name,
```
Comment (mdelapenya) [inline options.go:119]:
```suggestion
// prependHubRegistry represents a way to prepend a custom Hub registry to the image name,
```

**Comment by thaJeztah** [comment]
Think I have them all and changed them all back 😄

**Comment by mdelapenya** [comment]
Thanks for this! I'm enabling the CI now

**Comment by mdelapenya** [comment]
Oh you probably need to run `make lint` locally to detect and fix the deprecations

**Comment by mdelapenya** [comment]
> I should've read the docs

if the docs include reading the GH workflow, then you are right... otherwise it's my fault not having the CONTRIBUTING guidelines up-to-date 🙈

**Comment by thaJeztah** [comment]
Yup, looks like I missed one type 😂 I think it's ok now 🤞

**Comment by mdelapenya** [comment]
Perfect, lint passed! 🚀 Waiting for the CI to merge

**Comment by thaJeztah** [comment]
Hm... looks like some failures 🤔 fun!


```
run golangci-lint
  Running [/home/runner/golangci-lint-1.55.2-linux-amd64/golangci-lint run --out-format=github-actions --path-prefix=modules/artemis --verbose] in [/home/runner/work/testcontainers-go/testcontainers-go/modules/artemis] ...
  Error: undefined: testcontainers (typecheck)
  Error: undefined: testcontainers (typecheck)
  Error: undefined: testcontainers (typecheck)
  Error: c.PortEndpoint undefined (type *Container has no field or method PortEndpoint) (typecheck)
  Error: c.PortEndpoint undefined (type *Container has no field or method PortEndpoint) (typecheck)
```

**Comment by thaJeztah** [comment]
I don't see that locally; or at least, don't see something immediately wrong in  here; https://github.com/testcontainers/testcontainers-go/blob/2681ae411908eb9b8858d3e5b608598c647fe508/modules/artemis/artemis.go#L38

**Comment by thaJeztah** [comment]
Wondering if it's golangci-lint needing an update for newer Go versions 🤔 

```
golangci/golangci-lint-action@9d1e0624a798bb64f6c3cea93db47765312263dc
  with:
    version: v1.55.2
    working-directory: modules/artemis
    args: --verbose
    skip-cache: true
    github-token: ***
    only-new-issues: false
    skip-save-cache: false
    install-mode: binary
  env:
    TESTCONTAINERS_RYUK_DISABLED: false
    RYUK_CONNECTION_TIMEOUT: 60s
    RYUK_RECONNECTION_TIMEOUT: 10s
```

**Comment by mdelapenya** [comment]
Let me try to lint all modules locally 🙏

**Comment by mdelapenya** [comment]
Each submodule needed to mod-tidy again, I ran `make tidy-all` for that. This brings me the change to improve the contributing docs ASAP

**Comment by thaJeztah** [comment]
oh! missed that one; thanks!

**Comment by mdelapenya** [comment]
@thaJeztah https://github.com/testcontainers/testcontainers-go/pull/2586

File: modules/registry/registry.go [hunk 6]
```diff
@@ -135,7 +135,7 @@ func (c *RegistryContainer) PushImage(ctx context.Context, ref string) error {
 		return fmt.Errorf("failed to get image auth: %w", err)
 	}
 
-	pushOpts := types.ImagePushOptions{
+	pushOpts := image.PushOptions{
```
Comment (thaJeztah) [inline modules/registry/registry.go:138]: Still missed one! Maybe because separate module 🙈

Thanks for fixing

**Pull request merged**